In [1]:
import numpy as np
import pandas as pd

In [2]:
url = r'C:\Users\nuria\Proyecto_EDA\data\raw\Student Depression Dataset.csv'

### 1. Inspeccionar el tamaño y estructura de los datos

In [20]:
df = pd.read_csv(r'C:\Users\nuria\Proyecto_EDA\data\raw\Student Depression Dataset.csv')
df
# CGPA: Promedio de calificaciones u otros puntajes académicos.

id  Gender   Age           City Profession  Academic Pressure  \
0           2    Male  33.0  Visakhapatnam    Student                5.0   
1           8  Female  24.0      Bangalore    Student                2.0   
2          26    Male  31.0       Srinagar    Student                3.0   
3          30  Female  28.0       Varanasi    Student                3.0   
4          32  Female  25.0         Jaipur    Student                4.0   
...       ...     ...   ...            ...        ...                ...   
27896  140685  Female  27.0          Surat    Student                5.0   
27897  140686    Male  27.0       Ludhiana    Student                2.0   
27898  140689    Male  31.0      Faridabad    Student                3.0   
27899  140690  Female  18.0       Ludhiana    Student                5.0   
27900  140699    Male  27.0          Patna    Student                4.0   

       Work Pressure  CGPA  Study Satisfaction  Job Satisfaction  \
0                0.0  8.97                 2.0               0.0   
1                0.0  5.90                 5.0               0.0   
2                0.0  7.03                 5.0               0.0   
3                0.0  5.59                 2.0               0.0   
4                0.0  8.13                 3.0               0.0   
...              ...   ...                 ...               ...   
27896            0.0  5.75                 5.0               0.0   
27897            0.0  9.40                 3.0               0.0   
27898            0.0  6.61                 4.0               0.0   
27899            0.0  6.88                 2.0               0.0   
27900            0.0  9.24                 1.0               0.0   

          Sleep Duration Dietary Habits    Degree  \
0              5-6 hours        Healthy   B.Pharm   
1              5-6 hours       Moderate       BSc   
2      Less than 5 hours        Healthy        BA   
3              7-8 hours       Moderate       BCA   
4              5-6 hours       Moderate    M.Tech   
...                  ...            ...       ...   
27896          5-6 hours      Unhealthy  Class 12   
27897  Less than 5 hours        Healthy       MSc   
27898          5-6 hours      Unhealthy        MD   
27899  Less than 5 hours        Healthy  Class 12   
27900  Less than 5 hours        Healthy       BCA   

      Have you ever had suicidal thoughts ?  Work/Study Hours  \
0                                       Yes               3.0   
1                                        No               3.0   
2                                        No               9.0   
3                                       Yes               4.0   
4                                       Yes               1.0   
...                                     ...               ...   
27896                                   Yes               7.0   
27897                                    No               0.0   
27898                                    No              12.0   
27899                                   Yes              10.0   
27900                                   Yes               2.0   

       Financial Stress Family History of Mental Illness  Depression  
0                   1.0                               No           1  
1                   2.0                              Yes           0  
2                   1.0                              Yes           0  
3                   5.0                              Yes           1  
4                   1.0                               No           0  
...                 ...                              ...         ...  
27896               1.0                              Yes           0  
27897               3.0                              Yes           0  
27898               2.0                               No           0  
27899               5.0                               No           1  
27900               3.0                              Yes           1  

[27901 row

In [19]:
df.head()

id  Gender   Age           City Profession  Academic Pressure  \
0   2    Male  33.0  Visakhapatnam    Student                5.0   
1   8  Female  24.0      Bangalore    Student                2.0   
2  26    Male  31.0       Srinagar    Student                3.0   
3  30  Female  28.0       Varanasi    Student                3.0   
4  32  Female  25.0         Jaipur    Student                4.0   

   Work Pressure  CGPA  Study Satisfaction  Job Satisfaction  \
0            0.0  8.97                 2.0               0.0   
1            0.0  5.90                 5.0               0.0   
2            0.0  7.03                 5.0               0.0   
3            0.0  5.59                 2.0               0.0   
4            0.0  8.13                 3.0               0.0   

      Sleep Duration Dietary Habits   Degree  \
0          5-6 hours        Healthy  B.Pharm   
1          5-6 hours       Moderate      BSc   
2  Less than 5 hours        Healthy       BA   
3          7-8 hours       Moderate      BCA   
4          5-6 hours       Moderate   M.Tech   

  Have you ever had suicidal thoughts ?  Work/Study Hours  Financial Stress  \
0                                   Yes               3.0               1.0   
1                                    No               3.0               2.0   
2                                    No               9.0               1.0   
3                                   Yes               4.0               5.0   
4                                   Yes               1.0               1.0   

  Family History of Mental Illness  Depression  
0                               No           1  
1                              Yes           0  
2                              Yes           0  
3                              Yes           1  
4                               No           0

In [4]:
df.tail()

id  Gender   Age       City Profession  Academic Pressure  \
27896  140685  Female  27.0      Surat    Student                5.0   
27897  140686    Male  27.0   Ludhiana    Student                2.0   
27898  140689    Male  31.0  Faridabad    Student                3.0   
27899  140690  Female  18.0   Ludhiana    Student                5.0   
27900  140699    Male  27.0      Patna    Student                4.0   

       Work Pressure  CGPA  Study Satisfaction  Job Satisfaction  \
27896            0.0  5.75                 5.0               0.0   
27897            0.0  9.40                 3.0               0.0   
27898            0.0  6.61                 4.0               0.0   
27899            0.0  6.88                 2.0               0.0   
27900            0.0  9.24                 1.0               0.0   

          Sleep Duration Dietary Habits    Degree  \
27896          5-6 hours      Unhealthy  Class 12   
27897  Less than 5 hours        Healthy       MSc   
27898          5-6 hours      Unhealthy        MD   
27899  Less than 5 hours        Healthy  Class 12   
27900  Less than 5 hours        Healthy       BCA   

      Have you ever had suicidal thoughts ?  Work/Study Hours  \
27896                                   Yes               7.0   
27897                                    No               0.0   
27898                                    No              12.0   
27899                                   Yes              10.0   
27900                                   Yes               2.0   

       Financial Stress Family History of Mental Illness  Depression  
27896               1.0                              Yes           0  
27897               3.0                              Yes           0  
27898               2.0                               No           0  
27899               5.0                               No           1  
27900               3.0                              Yes           1

In [22]:
df.shape

(27901, 18)

Este conjunto de datos contiene 18 variables (columnas) y 27901 filas

In [25]:
# Verificar si hay jerarquías o estructuras anidadas
# Comprobar si hay listas o diccionarios en alguna columna
for column in df.columns:
    if df[column].apply(lambda x: isinstance(x, (list, dict))).any():
        print(f"La columna '{column}' tiene estructuras anidadas.")

No hay jerarquías ni estructuras anidadas

### 2. Verificar tipos de datos y nombres de columnas

In [21]:
df.columns

Index(['id', 'Gender', 'Age', 'City', 'Profession', 'Academic Pressure',
       'Work Pressure', 'CGPA', 'Study Satisfaction', 'Job Satisfaction',
       'Sleep Duration', 'Dietary Habits', 'Degree',
       'Have you ever had suicidal thoughts ?', 'Work/Study Hours',
       'Financial Stress', 'Family History of Mental Illness', 'Depression'],
      dtype='object')

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27901 entries, 0 to 27900
Data columns (total 18 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   id                                     27901 non-null  int64  
 1   Gender                                 27901 non-null  object 
 2   Age                                    27901 non-null  float64
 3   City                                   27901 non-null  object 
 4   Profession                             27901 non-null  object 
 5   Academic Pressure                      27901 non-null  float64
 6   Work Pressure                          27901 non-null  float64
 7   CGPA                                   27901 non-null  float64
 8   Study Satisfaction                     27901 non-null  float64
 9   Job Satisfaction                       27901 non-null  float64
 10  Sleep Duration                         27901 non-null  object 
 11  Di

|Columna| Descripción| Tipo de dato|
|---|---|---|
|id|id|int|
|Gender|Genero|str|
|Age|Edad|int|
|City|Ciudad|str|
|Profession|Profesión|str|
|Academic Pressure|Presión académica|float|
|Work Pressure|Presión laboral|float|
|CGPA|Promedio de calificaciones u otros puntajes académicos|float|
|Study Satisfaction|Satisfacción de estudio|float|
|Job Satisfaction|Satisfacción de trabajo|float|
|Sleep Duration|Duración de sueño|str|
|Dietary Habits|Dieta|str|
|Degree|Año de estudios|str|
|Have you ever had suicidal thoughts?|¿Alguna vez has tenidos pensamientos suicidas?|str|
|Work/Study Hours|Horas de trabajo/estudio|float|
|Financial Stress|Estres economico|float|
|Family History of Mental Illness|Antecedentes familiares de enfermedad mental|str|
|Depression|Depresión|int|

In [27]:
df

id  Gender   Age           City Profession  Academic Pressure  \
0           2    Male  33.0  Visakhapatnam    Student                5.0   
1           8  Female  24.0      Bangalore    Student                2.0   
2          26    Male  31.0       Srinagar    Student                3.0   
3          30  Female  28.0       Varanasi    Student                3.0   
4          32  Female  25.0         Jaipur    Student                4.0   
...       ...     ...   ...            ...        ...                ...   
27896  140685  Female  27.0          Surat    Student                5.0   
27897  140686    Male  27.0       Ludhiana    Student                2.0   
27898  140689    Male  31.0      Faridabad    Student                3.0   
27899  140690  Female  18.0       Ludhiana    Student                5.0   
27900  140699    Male  27.0          Patna    Student                4.0   

       Work Pressure  CGPA  Study Satisfaction  Job Satisfaction  \
0                0.0  8.97                 2.0               0.0   
1                0.0  5.90                 5.0               0.0   
2                0.0  7.03                 5.0               0.0   
3                0.0  5.59                 2.0               0.0   
4                0.0  8.13                 3.0               0.0   
...              ...   ...                 ...               ...   
27896            0.0  5.75                 5.0               0.0   
27897            0.0  9.40                 3.0               0.0   
27898            0.0  6.61                 4.0               0.0   
27899            0.0  6.88                 2.0               0.0   
27900            0.0  9.24                 1.0               0.0   

          Sleep Duration Dietary Habits    Degree  \
0              5-6 hours        Healthy   B.Pharm   
1              5-6 hours       Moderate       BSc   
2      Less than 5 hours        Healthy        BA   
3              7-8 hours       Moderate       BCA   
4              5-6 hours       Moderate    M.Tech   
...                  ...            ...       ...   
27896          5-6 hours      Unhealthy  Class 12   
27897  Less than 5 hours        Healthy       MSc   
27898          5-6 hours      Unhealthy        MD   
27899  Less than 5 hours        Healthy  Class 12   
27900  Less than 5 hours        Healthy       BCA   

      Have you ever had suicidal thoughts ?  Work/Study Hours  \
0                                       Yes               3.0   
1                                        No               3.0   
2                                        No               9.0   
3                                       Yes               4.0   
4                                       Yes               1.0   
...                                     ...               ...   
27896                                   Yes               7.0   
27897                                    No               0.0   
27898                                    No              12.0   
27899                                   Yes              10.0   
27900                                   Yes               2.0   

       Financial Stress Family History of Mental Illness  Depression  
0                   1.0                               No           1  
1                   2.0                              Yes           0  
2                   1.0                              Yes           0  
3                   5.0                              Yes           1  
4                   1.0                               No           0  
...                 ...                              ...         ...  
27896               1.0                              Yes           0  
27897               3.0                              Yes           0  
27898               2.0                               No           0  
27899               5.0                               No           1  
27900               3.0                              Yes           1  

[27901 row

- Memoria: 3.8+ MB
- La edad sale como flotante
- Sleep durantion sale tipo object
- Todos los textos salen tipo object pero me llaman la atnción 'Gender' y 'Have you ever had suicidal thoughts?'

No tenemos valores nulos

### 3. Identificar valores duplicados o errores evidentes

In [23]:
duplicados = df.duplicated()
duplicados.head(10)

0    False
1    False
2    False
3    False
4    False
5    False
6    False
7    False
8    False
9    False
dtype: bool

In [16]:
# Observamos que tenga valores missing
df[df.duplicated(keep = False)] 

Empty DataFrame
Columns: [id, Gender, Age, City, Profession, Academic Pressure, Work Pressure, CGPA, Study Satisfaction, Job Satisfaction, Sleep Duration, Dietary Habits, Degree, Have you ever had suicidal thoughts ?, Work/Study Hours, Financial Stress, Family History of Mental Illness, Depression]
Index: []

No tenemos valores duplicados exactos ni valores missing

In [26]:
df.isnull()

id  Gender    Age   City  Profession  Academic Pressure  \
0      False   False  False  False       False              False   
1      False   False  False  False       False              False   
2      False   False  False  False       False              False   
3      False   False  False  False       False              False   
4      False   False  False  False       False              False   
...      ...     ...    ...    ...         ...                ...   
27896  False   False  False  False       False              False   
27897  False   False  False  False       False              False   
27898  False   False  False  False       False              False   
27899  False   False  False  False       False              False   
27900  False   False  False  False       False              False   

       Work Pressure   CGPA  Study Satisfaction  Job Satisfaction  \
0              False  False               False             False   
1              False  False               False             False   
2              False  False               False             False   
3              False  False               False             False   
4              False  False               False             False   
...              ...    ...                 ...               ...   
27896          False  False               False             False   
27897          False  False               False             False   
27898          False  False               False             False   
27899          False  False               False             False   
27900          False  False               False             False   

       Sleep Duration  Dietary Habits  Degree  \
0               False           False   False   
1               False           False   False   
2               False           False   False   
3               False           False   False   
4               False           False   False   
...               ...             ...     ...   
27896           False           False   False   
27897           False           False   False   
27898           False           False   False   
27899           False           False   False   
27900           False           False   False   

       Have you ever had suicidal thoughts ?  Work/Study Hours  \
0                                      False             False   
1                                      False             False   
2                                      False             False   
3                                      False             False   
4                                      False             False   
...                                      ...               ...   
27896                                  False             False   
27897                                  False             False   
27898                                  False             False   
27899                                  False             False   
27900                                  False             False   

       Financial Stress  Family History of Mental Illness  Depression  
0                 False                             False       False  
1                 False                             False       False  
2                 False                             False       False  
3                 False                             False       False  
4                 False                             False       False  
...                 ...                               ...         ...  
27896             False                             False       False  
27897             False                             False       False  
27898             False                             False       False  
27899             False                             False       False  
27900             False                             False       False  

[27901 rows x 18 columns]

In [10]:
df.nunique()

id                                       27901
Gender                                       2
Age                                         34
City                                        52
Profession                                  14
Academic Pressure                            6
Work Pressure                                3
CGPA                                       332
Study Satisfaction                           6
Job Satisfaction                             5
Sleep Duration                               5
Dietary Habits                               4
Degree                                      28
Have you ever had suicidal thoughts ?        2
Work/Study Hours                            13
Financial Stress                             5
Family History of Mental Illness             2
Depression                                   2
dtype: int64

## Observaciones
- El conjunto de datos se compone de 18 columnas (variables) y 27901 filas
- Utiliza un espacio de memoria de 3.5+ MB
- No hay valores nulos
- No hay valores duplicados
- No hay datos faltantes
- Hay algunos tipos de dato que probablemente tenga que cambiar
- No hay jerarquías o estructuras anidadas

In [ ]:
df_final.to_parquet(data/processed/futbol_limpio.parquet)